File: runelite/runelite-client/src/main/java/net/runelite/client/plugins/gamedataserver/model/exchange/ExchangeSlot.java
package net.runelite.client.plugins.gamedataserver.model.exchange;

import lombok.Builder;
import lombok.Data;

@Builder
@Data
public class ExchangeSlot {
	private int position;
	private int itemId;
	private int price;
	private int quantityTransacted;
	private int totalQuantity;
	private ExchangeSlotState state;

	public static ExchangeSlot asEmpty(int position) {
		return ExchangeSlot.builder()
			.position(position)
			.itemId(-1)
			.price(-1)
			.quantityTransacted(-1)
			.totalQuantity(-1)
			.state(ExchangeSlotState.EMPTY)
			.build();
	}
}

File: autotrader/src/clients/gds/models/exchange/exchange_slot.py
from dataclasses import dataclass

from clients.gds.models.exchange.exchange_slot_state import ExchangeSlotState


@dataclass
class ExchangeSlot:
    position: int
    item_id: int
    price: int
    quantity_transacted: int
    total_quantity: int
    state: ExchangeSlotState
